### PR TITLE
Adjust JsonRPCPlugin.cs Exception Handling

### PR DESCRIPTION
**Pull request by taooceros**
#632 discussion

1. Expose Exception from JsonRPCPlugin so users may notice there's an exception thrown.
2. Don't stop result parse when stderr contains message.

### CODE REVIEW

File: Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs [hunk 1]
```diff
@@ -249,12 +228,14 @@ protected async Task<Stream> ExecuteAsync(ProcessStartInfo startInfo, Cancellati
...
                 var buffer = BufferManager.GetStream();
-                
+
                 token.Register(() =>
                 {
                     // ReSharper disable once AccessToModifiedClosure
                     // Manually Check whether disposed
+                    // ReSharper disable once AccessToDisposedClosure
                     if (!disposed && !process.HasExited)
+                        // ReSharper disable once AccessToDisposedClosure
```
Comment (jjw24) [inline Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs:238]: i remember i deleted these two before, how come they are readded?

File: Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs [hunk 2]
```diff
@@ -249,12 +228,14 @@ protected async Task<Stream> ExecuteAsync(ProcessStartInfo startInfo, Cancellati
                 await using var source = process.StandardOutput.BaseStream;
 
                 var buffer = BufferManager.GetStream();
-                
+
                 token.Register(() =>
                 {
                     // ReSharper disable once AccessToModifiedClosure
                     // Manually Check whether disposed
+                    // ReSharper disable once AccessToDisposedClosure
```
Comment (jjw24) [inline Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs:236]:
```suggestion
```

File: Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs [hunk 3]
```diff
@@ -249,12 +228,13 @@ protected async Task<Stream> ExecuteAsync(ProcessStartInfo startInfo, Cancellati
...
 
                 var buffer = BufferManager.GetStream();
-                
+
                 token.Register(() =>
                 {
                     // ReSharper disable once AccessToModifiedClosure
                     // Manually Check whether disposed
                     if (!disposed && !process.HasExited)
+                        // ReSharper disable once AccessToDisposedClosure
```
Comment (jjw24) [inline Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs:237]:
```suggestion
```

**Review by jjw24 — APPROVED** [review]

File: Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs (hunk 1 above)
Comment (taooceros) [reply Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs]: Oh I add it to suppress the warning. Sorry, good to remove them

Comment (jjw24) [reply Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs]: Oh ok, sorry I don't use resharper so didn't know what they r for, if required please add them back in.

Comment (taooceros) [reply Flow.Launcher.Core/Plugin/JsonRPCPlugin.cs]: > Oh ok, sorry I don't use resharper so didn't know what they r for, if required please add them back in.

No just the yellow wavy line a bit annoying.

**Pull request merged**